src/components/uikit/Button/Button.tsx

```diff
@@ -1,23 +1,60 @@
-import { cva } from "~/styled-system/css";
-import { styled } from "~/styled-system/jsx";
+import { ComponentProps } from "react";
+import { cva, type RecipeVariantProps } from "~/styled-system/css";
 
 export const button = cva({
   base: {
-    fontWeight: "medium",
-    borderRadius: "md",
+    borderRadius: "3xl",
+    color: "slate.950",
+    border: "0.5px solid",
+    cursor: "pointer",
   },
   variants: {
-    status: {
+    variant: {
       default: {
-        color: "white",
-        bg: "gray.500",
+        bg: "yellow.300",
+        borderColor: "yellow.300",
+        fontWeight: "bold",
+        _hover: {
+          bg: "yellow.200",
+          borderColor: "yellow.200",
+        },
       },
-      success: {
-        color: "white",
-        bg: "green.500",
+      unfilled: {
+        bg: "slate.50",
+        borderColor: "slate.950",
+        fontWeight: "normal",
+        _hover: {
+          textDecoration: "underline",
+        },
+      },
+    },
+    size: {
+      sm: {
+        fontSize: "md",
+        lineHeight: "normal",
+        px: "25px",
+        py: "8px",
       },
     },
   },
 });
 
-export const Button = styled("button", button);
+type ButtonProps = {
+  icon?: React.ReactNode;
+} & RecipeVariantProps<typeof button> &
+  ComponentProps<"button">;
+
+export function Button({
+  icon = null,
+  variant = "default",
+  size = "sm",
+  children,
+  ...restProps
+}: ButtonProps) {
+  return (
+    <button className={button({ variant, size })} {...restProps}>
+      {icon}
```

Comment:
Надо протестить вариант с иконкой, добавить тоже его в сторибук
Кажется, лучше будет обернуть иконку в `styled.div`, будет проще с ней работать
```suggestion
      {icon && <styled.div display="inline-block">{icon}</styled.div>}
```

Reply:
вообще, в идеале - отдельные свойства isLoading, isDisabled, но об этом лучше говорить с дизайнером, сами придумать, конечно, можем, главное чтобы она потом эти кнопки использовала

Reply:
насчет вида `isDisabled` да. это в целом состояние кнопки, которая зависит 100% от дизайна.

насчет загрузки не уверен. кнопка это кнопка, поэтому кажется лодер лучше рядом класть и делать его отдельным компонентом. тогда состояние кнопки будет регулироваться снаружи типа `disabled={isLoading}`

Reply:
согласен, давай сделаем компонент Spinner, в качестве референса можно попробовать взять вот это https://loading.io/css/